_posts/2019-04-30-accessing-kafka-part-3.md

```diff
@@ -0,0 +1,165 @@
+* _[Part 2 - Node Ports](https://strimzi.io/2019/04/23/accessing-kafka-part-2.html)_
+* _Part 3 - OpenShift Routes (this post)_
+
+# OpenShift Routes
+
+Routes are an OpenShift concept for exposing services to the outside of the OpenShift platform.
+Routes handle both data routing as well as DNS resolution.
+DNS resolution is usually handled using [wildcard DNS entries](https://en.wikipedia.org/wiki/Wildcard_DNS_record).
+That allows OpenShift to assign each route its own DNS name which is based on the wildcard entry.
+Users do not have to anything special to handle the DNS records.
+But don't worry, when you don't own any domains where you can setup the wildcard entires, it can use services such as [nip.ip](https://nip.io/) for the wildcard DNS routing.
+Data routing is done using the [HAProxy](https://www.haproxy.org) load balancer which serves as the router behind the routes.
+
+The main use-case of the router is HTTP(S) routing.
+The routes are able to do path based routing of HTTP and HTTPS (with TLS termination) traffic.
+In this mode, the HTTP requests will be routed to different services based on the request path.
+However, since the Kafka protocol is not based on HTTP, the HTTP features are not very useful for Strimzi and Kafka brokers.
+
+But luckily the routes can be also used for TLS passthrough.
+In this mode, it uses TLS SNI to determine the service to which the traffic should be routed and passes the TLS connection to the service (and eventually to the pod backing the service) without decoding it.
+This mode is what Strimzi is using for exposing Kafka.
+
+If you want to learn more about OpenShift Routes, check the [OpenShift documentation](https://docs.openshift.com/container-platform/3.11/architecture/networking/routes.html).
+
+# Exposing Kafka using OpenShift Routes
+
+Exposing Kafka using OpenShift Routes is probably the easiest of all the available options.
+All you need to do is to configure it in the Kafka custom resource.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+    listeners:
+      # ...
+      external:
+        type: route
+    # ...
+```
+
+And Strimzi Kafka Operator and OpenShift will take care of the rest.
+To provide access to the individual brokers, we use the same tricks as we use with node ports and which were already described in the [previous blog post](https://strimzi.io/2019/04/23/accessing-kafka-part-2.html).
+We create a dedicated service for each of the brokers.
+These will be used to address the individual brokers directly.
+Apart from that we will also use one service for the bootstrapping of the clients.
+This service would round-robin between all Kafka brokers.
+
+But unlike when using node ports, these services will be only regular Kubernetes services of the `clusterIP` type.
+Instead, the Strimzi Kafka operator will also create a `Route` resource for each of these services.
```

Comment:
I find the instead a bit confusing here, because it's not really instead of the preceding sentence, but instead of the comparison with node ports.

```suggestion
The Strimzi Kafka operator will also create a `Route` resource for each of these services.
```